tests/test_hexary_trie.py

```diff
@@ -1013,3 +1013,39 @@ def test_traverse_from_partial_path(
     assert exc.nibbles_traversed == path_to_node
     assert exc.node.sub_segments == sub_segments
     assert exc.node.value == node_val
+
+
+def test_squash_a_pruning_trie_keeps_unchanged_short_root_node():
+    db = {}
+    trie = HexaryTrie(db, prune=True)
+    trie[b'any'] = b'short'
+    root_hash = trie.root_hash
+    with trie.squash_changes() as trie_batch:
+        trie_batch[b'any'] = b'short'
+        assert trie.root_hash == root_hash
+        assert root_hash in trie_batch.db
+        assert root_hash in db
+    assert trie.root_hash == root_hash
+    assert root_hash in trie_batch.db
```

Comment:
This is the assertion that was failing before the bugfix.

Reply:
Maybe a basic context manager detail, but curious why this would fail when the same assertion three lines above might've passed.